src/converter/one/heatmap/MethodCache.java

```diff
@@ -6,20 +6,22 @@
 package one.heatmap;
 
 import java.util.Arrays;
+import java.util.LinkedList;
 
 import one.convert.Index;
 import one.convert.JfrConverter;
 import one.jfr.Dictionary;
 
+// TODO Rename to FrameDescCache
 public class MethodCache {
     private final JfrConverter converter;
     private final Index<String> symbolTable = new Index<>(String.class, "", 32768);
-    private final Index<Method> methodIndex = new Index<>(Method.class, new Method(symbolTable.index("all"), 0), 32768);
+    private final Index<FrameDesc> methodIndex = new Index<>(FrameDesc.class, new FrameDesc(symbolTable.index("all"), 0), 32768);
 
-    private final Method[] nearCache = new Method[256 * 256];
+    private final FrameDesc[] nearCache = new FrameDesc[256 * 256];
     // It should be better to create dictionary with linked methods instead of open addressed hash table
     // but in most cases all methods should fit nearCache, so less code is better
-    private final Dictionary<Method> farMethods = new Dictionary<>(1024);
+    private final Dictionary<LinkedList<FrameDesc>> farMethods = new Dictionary<>(1024);
```

Comment:
Dictionary of linked list, addressed by method id? What is the intention here? Why is this better than the hash table?

Reply:
If you mean why `Dictionary<LinkedList>` instead of `HashMap<LinkedList>`, I'd be happy to replace all the occurrences of `Dictionary` with `HashMap`, I don't see other advantages than no boxing. Could you weigh in @apangin ?

If you mean why `LinkedList`, the linked list was already there, it was implemented with a field `FrameDesc.next`